lwgenerate/aws_controltower/aws_controltower.go

```diff
@@ -0,0 +1,455 @@
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.ExternalIdLength = length
+	}
+}
+
+func WithWaitTime(waitTime int) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.WaitTime = waitTime
+	}
+}
+
+func WithTags(tags map[string]string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.Tags = tags
+	}
+}
+
+func WithKmsKeyArn(arn string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.KmsKeyArn = arn
+	}
+}
+
+func WithExisitingIamRole(arn string, name string, externalID string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.IamRoleArn = arn
+		c.IamRoleExternalID = externalID
+		c.IamRoleName = name
+		c.UseExistingIamRole = true
+	}
+}
+
+func WithPrefix(prefix string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.Prefix = prefix
+	}
+}
+
+func WithSqsQueueName(name string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.SqsQueueName = name
+	}
+}
+
+func WithOrgAccountMappings(mapping OrgAccountMapping) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.OrgAccountMappings = mapping
+	}
+}
+
+func WithLaceworkIntegrationName(name string) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.LaceworkIntegrationName = name
+	}
+}
+
+func WithLaceworkOrgLevel() AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.LaceworkOrganizationLevel = true
+	}
+}
+
+func WithSubaccounts(subaccounts ...AwsSubAccount) AwsControlTowerTerraformModifier {
+	return func(c *GenerateAwsControlTowerTfConfigurationArgs) {
+		c.SubAccounts = subaccounts
+	}
+}
+
+func createCloudTrailControlTower(args *GenerateAwsControlTowerTfConfigurationArgs) (*hclwrite.Block, error) {
+	attributes := map[string]interface{}{}
+	modDetails := []lwgenerate.HclModuleModifier{
+		lwgenerate.HclModuleWithVersion(lwgenerate.AwsCloudTrailControlTowerVersion)}
+
+	//required args
+	if args.S3BucketArn != "" {
+		attributes["s3_bucket_arn"] = args.S3BucketArn
+	}
+	if args.SNSTopicArn != "" {
+		attributes["sns_topic_arn"] = args.SNSTopicArn
+	}
+
+	// optional args
+	if args.LaceworkAccountID != "" {
+		attributes["lacework_aws_account_id"] = args.LaceworkAccountID
+	}
+
+	if args.CrossAccountPolicyName != "" {
+		attributes["cross_account_policy_name"] = args.CrossAccountPolicyName
+	}
+	if args.EnableLogFileValidation {
+		attributes["enable_log_file_validation"] = args.EnableLogFileValidation
+	}
+	if args.ExternalIdLength != 0 {
+		attributes["external_id_length"] = args.ExternalIdLength
+	}
+	if args.WaitTime != 0 {
+		attributes["wait_time"] = args.WaitTime
+	}
+	if args.Tags != nil {
```

Comment:
If this variable was being set with a Cobra default, wouldn't the de something other than `nil`.

I like to prevent the presence of `nil` or `null` in code.  Maybe GO is ok with this though.

Reply:
Also see IDE flagging this with `Invalid operation: args.Tags != nil (the operator != is not defined on map[string]string)`

Reply:
I've no idea why the ide is flagging that, a map can be nil...
https://go.dev/play/p/IMayAh8yztA

I don't see any issues with this.